rust/processor/src/models/fungible_asset_models/v2_fungible_asset_balances.rs

```diff
@@ -62,6 +69,89 @@ pub struct CurrentFungibleAssetBalance {
     pub token_standard: String,
 }
 
+#[derive(Clone, Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize, Default)]
+#[diesel(primary_key(storage_id))]
+#[diesel(table_name = current_unified_fungible_asset_balances)]
+#[diesel(treat_none_as_null = true)]
+pub struct CurrentUnifiedFungibleAssetBalance {
+    pub storage_id: String,
+    pub owner_address: String,
+    // metadata address for (paired) Fungible Asset
+    pub asset_type: String,
+    pub coin_type: Option<String>,
+    pub is_primary: Option<bool>,
+    pub is_frozen: bool,
+    pub amount_v1: Option<BigDecimal>,
+    pub amount_v2: Option<BigDecimal>,
+    pub last_transaction_version_v1: Option<i64>,
+    pub last_transaction_version_v2: Option<i64>,
+    pub last_transaction_timestamp_v1: Option<chrono::NaiveDateTime>,
+    pub last_transaction_timestamp_v2: Option<chrono::NaiveDateTime>,
+}
+
+fn get_paired_metadata_address(coin_type_name: &str) -> anyhow::Result<String> {
+    let mut preimage = APT_METADATA_ADDRESS.as_bytes().to_vec();
+    preimage.extend(coin_type_name.as_bytes());
+    preimage.push(0xFE);
+    Ok(String::from_utf8(sha3_256(&preimage).to_vec())?)
+}
+
+fn get_primary_fungible_store_address(
+    owner_address: &str,
+    metadata_address: &str,
+) -> anyhow::Result<String> {
+    let mut preimage = hex_to_raw_bytes(owner_address)?;
```

Comment:
is the address padded with 0 if not full 64 long.

Reply:
I follow the existing logic.